From 1b5762df2a6a7dbad478006349ae81e011b0d33d Mon Sep 17 00:00:00 2001
From: Allen Xu <b45815@freescale.com>
Date: Wed, 24 Sep 2014 15:00:18 -0500
Subject: [PATCH 1169/1691] ENGR00333048 ARM: dts: add new dts for gpmi and
 weim on imx6q sabreauto

commit 1b5762df2a6a7dbad478006349ae81e011b0d33d from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

new dts file to support gpmi/weim on imx6q sabreauto board.

Signed-off-by: Allen Xu <b45815@freescale.com>
(cherry picked and merge from commit d51a5439d7b996a26e842c8d4034668e4c7959ab)
---
 arch/arm/boot/dts/Makefile                      |  1 +
 arch/arm/boot/dts/imx6q-sabreauto-gpmi-weim.dts | 52 +++++++++++++++++++++++++
 2 files changed, 53 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6q-sabreauto-gpmi-weim.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 9208893..d834fdd 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -300,6 +300,7 @@ dtb-$(CONFIG_SOC_IMX6Q) += \
 	imx6q-phytec-pbab01.dtb \
 	imx6q-rex-pro.dtb \
 	imx6q-sabreauto.dtb \
+	imx6q-sabreauto-gpmi-weim.dtb \
 	imx6qp-sabreauto.dtb \
 	imx6qp-sabresd.dtb \
 	imx6qp-sabresd-btwifi.dtb \
diff --git a/arch/arm/boot/dts/imx6q-sabreauto-gpmi-weim.dts b/arch/arm/boot/dts/imx6q-sabreauto-gpmi-weim.dts
new file mode 100644
index 0000000..269d362
--- /dev/null
+++ b/arch/arm/boot/dts/imx6q-sabreauto-gpmi-weim.dts
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6q-sabreauto.dts"
+
+&ecspi1 {
+	/* pin conflict with weim */
+	status = "disabled";
+};
+
+/* remove #if 0 after flexcan is ready */
+#if 0
+&flexcan2 {
+	/* max7310_c on i2c3 is gone */
+	status = "disabled";
+};
+#endif
+
+&gpmi {
+	status = "okay";
+};
+
+&i2c3 {
+	/* pin conflict with weim */
+	status = "disabled";
+};
+
+&uart3 {
+	/* pin conflict with gpmi and weim */
+	status = "disabled";
+};
+
+&usbh1 {
+	/* max7310_b on i2c3 is gone */
+	status = "disabled";
+};
+
+&usbotg {
+	/* max7310_c on i2c3 is gone */
+	status = "okay";
+	dr_mode = "peripheral";
+};
+
+&weim {
+	pinctrl-assert-gpios = <&gpio5 4 GPIO_ACTIVE_LOW>;
+	status = "okay";
+};
-- 
1.9.1

